src/python/pants/init/engine_initializer.py

```diff
@@ -196,30 +198,36 @@ def __init__(self, invalid_goals):
       )
       self.invalid_goals = invalid_goals
 
-  def run_goal_rules(self, options_bootstrapper, options, goals, target_roots):
+  def run_goal_rules(
```

Comment:
FYI this is what I was looking for! This function is how every V2 goal gets the original `AddressSpecs`.